### ISSUE TITLE
False positive on `super.catch()` method

### ISSUE THREAD

**Issue by OoDeLally** [eslint-community/eslint-plugin-promise#599]
NestJs exposes a `BaseExceptionFilter` with a `catch()` method.

When extending this class, it creates false positives:

```
Promise.catch() requires 1 argument, but received 2eslint[promise/valid-params](https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/valid-params.md)

Prefer await to then()/catch()/finally().eslint[promise/prefer-await-to-then](https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/prefer-await-to-then.md)
```

# Simple reproduction

```ts

class Mother {
  catch() {}
}

class Child extends Mother {
  catch() {
    super.catch(); // Eslint warning
  }
}
```

It seems like `catch` is an officially reserved word in JS/TS, so you could consider that the mistake is on NestJs side for using `catch()` as a method. However, if it is easy to fix `eslint-plugin-promise` to avoid this, it would make everyone happy 🙏
